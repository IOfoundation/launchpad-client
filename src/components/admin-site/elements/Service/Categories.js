export const Categories = {
  leftColumn: [
    {
      key: 'financialManagement',
      label: 'Financial Management',
    },
    {
      key: 'capital',
      label: 'Capital',
    },
    {
      key: 'legalServices',
      label: 'Legal Services',
    },
    {
      key: 'marketingSales',
      label: 'Marketing/Sales',
    },
    {
      key: 'networking',
      label: 'Networking',
    },
    {
      key: 'manufacturingLogistics',
      label: 'Manufacturing/Logistics',
    },
    {
      key: 'procurement',
      label: 'Procurement',
    },
  ],
  rightColumn: [
    {
      key: 'planningManagement',
      label: 'Planning/Management',
    },
    {
      key: 'RDCommercialization',
      label: 'R&D/Commercialization',
    },
    {
      key: 'regulatoryCompliance',
      label: 'Regulatory Compliance',
    },
    {
      key: 'physicalSpace',
      label: 'Physical Space',
    },
    {
      key: 'mentoringCounseling',
      label: 'Mentoring/Counseling',
    },
    {
      key: 'humanResourcesWorkforceDevelopment',
      label: 'Human Resources & Workforce Development',
    },
  ],
};
