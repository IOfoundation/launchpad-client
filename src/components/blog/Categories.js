import React from 'react';
import Category from './Category';

const Categories = () => {
  return (
    <div className="blog-categories">
      <h3>{'Categories'}</h3>
      <ul className="blog-categories">
        <Category to="#" label="Front Page" />
        <Category to="#" label="Category Title" />
        <Category to="#" label="Another Category" />
        <Category to="#" label="Category Title" />
        <Category to="#" label="Category Title" />
        <Category to="#" label="Category Title" />
        <Category to="#" label="Category Title" />
        <Category to="#" label="Category Title" />
        <Category to="#" label="Category Title" />
      </ul>
      <button className="outline-button">{'More Categories'}</button>
    </div>
  );
};

export default Categories;
