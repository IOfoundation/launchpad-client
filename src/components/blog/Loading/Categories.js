import React, {Fragment} from 'react';

const CategoriesLoading = () => {
  return (
    <Fragment>
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
      <div className="blog-categories-loading__item animated-background" />
    </Fragment>
  );
};

export default CategoriesLoading;
