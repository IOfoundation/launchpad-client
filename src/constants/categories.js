const Categories = {
  BUSINESS_SERVICES: 'Categories',
  BUSINESS_TYPE: 'Business Type',
  STAGE: 'Business Stage',
  COMMUNITY: 'Underserved Communities',
  INDUSTRY: 'Industry',
};

export default Categories;
