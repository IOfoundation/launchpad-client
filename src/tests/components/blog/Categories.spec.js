import React from 'react';
import {mount} from 'enzyme';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {categories} from './categories.mock';

import Categories from '../../../components/blog/Categories';

const initialState = {
  blogs: {
    categories: [...categories],
    category: 'front page',
    featuredPosts: [],
    noResults: false,
    organizationPosts: [],
    page: 1,
    posts: [],
    totalPages: 0,
  },
};
const mockStore = configureStore([thunk]);
const props = {};
let store;
let wrapper;

describe('<Categories />', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Categories {...props} />
      </Provider>
    );
  });

  it('renders snapshot of Categories', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('display more categories', () => {
    const button = wrapper.find('button.button-outline');

    expect(button.text()).toBe('More Categories');
    button.simulate('click');

    expect(button.text()).toBe('View Less');
  });

  it('filter by categorie with default views', () => {
    const category = wrapper.find('.blog-categories__list__item');

    expect(store.getActions().length).toBe(1);
    category.at(2).simulate('click');

    expect(store.getActions().length).toBe(2);
  });

  it('filter by categorie with view more on true', () => {
    const category = wrapper.find('.blog-categories__list__item');
    const button = wrapper.find('button.button-outline');

    expect(store.getActions().length).toBe(1);
    button.simulate('click');
    category.at(2).simulate('click');

    expect(store.getActions().length).toBe(2);
  });
});
