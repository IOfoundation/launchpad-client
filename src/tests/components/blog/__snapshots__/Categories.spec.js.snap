// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders snapshot of Categories 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Categories)>
    <Categories
      actions={
        Object {
          "getAllPosts": [Function],
          "getCategories": [Function],
          "getFeaturedPost": [Function],
          "getFeaturedPostById": [Function],
          "getPostById": [Function],
          "noResults": [Function],
          "setCategory": [Function],
        }
      }
      categories={
        Array [
          Object {
            "id": 10,
            "name": "health",
            "taggings_count": 0,
          },
          Object {
            "id": 11,
            "name": "work",
            "taggings_count": 0,
          },
          Object {
            "id": 16,
            "name": "spanish",
            "taggings_count": 0,
          },
          Object {
            "id": 18,
            "name": "motorcycle",
            "taggings_count": 0,
          },
          Object {
            "id": 21,
            "name": "music",
            "taggings_count": 0,
          },
          Object {
            "id": 25,
            "name": "decorations",
            "taggings_count": 0,
          },
          Object {
            "id": 1,
            "name": "featured",
            "taggings_count": 6,
          },
          Object {
            "id": 2,
            "name": "front page",
            "taggings_count": 6,
          },
          Object {
            "id": 17,
            "name": "home",
            "taggings_count": 2,
          },
          Object {
            "id": 24,
            "name": "outdoors",
            "taggings_count": 2,
          },
          Object {
            "id": 20,
            "name": "clothes",
            "taggings_count": 3,
          },
          Object {
            "id": 13,
            "name": "rock",
            "taggings_count": 1,
          },
          Object {
            "id": 14,
            "name": "rap",
            "taggings_count": 1,
          },
          Object {
            "id": 6,
            "name": "drink",
            "taggings_count": 1,
          },
          Object {
            "id": 12,
            "name": "basketball",
            "taggings_count": 1,
          },
          Object {
            "id": 15,
            "name": "country",
            "taggings_count": 3,
          },
          Object {
            "id": 19,
            "name": "shoes",
            "taggings_count": 1,
          },
          Object {
            "id": 9,
            "name": "entertainment",
            "taggings_count": 2,
          },
          Object {
            "id": 3,
            "name": "soccer",
            "taggings_count": 1,
          },
          Object {
            "id": 4,
            "name": "games",
            "taggings_count": 4,
          },
          Object {
            "id": 7,
            "name": "programming",
            "taggings_count": 3,
          },
          Object {
            "id": 22,
            "name": "fashion",
            "taggings_count": 3,
          },
          Object {
            "id": 5,
            "name": "food",
            "taggings_count": 3,
          },
          Object {
            "id": 23,
            "name": "wear",
            "taggings_count": 2,
          },
          Object {
            "id": 8,
            "name": "study",
            "taggings_count": 4,
          },
        ]
      }
      categorySelected="front page"
    >
      <div
        className="blog-categories"
      >
        <h3
          className="blog-categories__title text-bold"
        >
          <i
            className="material-icons blog-categories__title__icon"
          >
            label
          </i>
          <span>
            Categories
          </span>
        </h3>
        <ul
          className="blog-categories__list"
        >
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="10"
            label="health"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              health
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="11"
            label="work"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              work
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="16"
            label="spanish"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              spanish
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="18"
            label="motorcycle"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              motorcycle
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="21"
            label="music"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              music
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="25"
            label="decorations"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              decorations
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="1"
            label="featured"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              featured
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={true}
            key="2"
            label="front page"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item blog-categories__list__selected"
              onClick={[Function]}
            >
              front page
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="17"
            label="home"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              home
            </li>
          </Category>
          <Category
            className="blog-categories__list__item"
            clicked={[Function]}
            isSelected={false}
            key="24"
            label="outdoors"
          >
            <li
              className="blog-categories capitalize blog-categories__list__item"
              onClick={[Function]}
            >
              outdoors
            </li>
          </Category>
        </ul>
        <button
          className="button-outline view-more text-bold"
          onClick={[Function]}
        >
          More Categories
        </button>
      </div>
    </Categories>
  </Connect(Categories)>
</Provider>
`;
